src/activity_item_handling.cpp

```diff
@@ -649,6 +650,73 @@ void activity_on_turn_move_items()
     }
 }
 
+static void move_item( item &it, int quantity, const tripoint &src, const tripoint &dest )
+{
+    item leftovers = it;
+
+    if( quantity != 0 && it.count_by_charges() ) {
+        // Reinserting leftovers happens after item removal to avoid stacking issues.
+        leftovers.charges = it.charges - quantity;
+        if( leftovers.charges > 0 ) {
+            it.charges = quantity;
+        }
+    } else {
+        leftovers.charges = 0;
+    }
+
+    // Check that we can pick it up.
+    if( !it.made_of( LIQUID ) ) {
+        g->u.mod_moves( -Pickup::cost_to_move_item( g->u, it ) );
```

Comment:
This needs to factor in cost to shuttle player between loot and target pile.
If you want to reduce trips it can be one trip per available inventory capacity, but then it needs to include inventory insertion/removal time as well.

Reply:
Fixed here https://github.com/CleverRaven/Cataclysm-DDA/pull/24488/commits/774ee224f37050029e237f64c0223d2ccb646687 (?)